File: src/game/graphics/Sprite.java
package game.graphics;

/**
 * Location of the image in the atlas
 */
public class Sprite {
    public final int startX;
    public final int startY;
    public final int endX;
    public final int endY;
    public final int width;
    public final int height;

    public Sprite(int startX, int startY, int width, int height) {
        this.startX = startX;
        this.startY = startY;
        this.width = width;
        this.height = height;
        endX = startX + width;
        endY = startY + height;
    }
}

File: src/game/graphics/Sprite.kt
package game.graphics

/**
 * Location of the image in the atlas
 */
class Sprite(val startX: Int, val startY: Int, val width: Int, val height: Int) {
    val endX: Int
    val endY: Int

    init {
        endX = startX + width
        endY = startY + height
    }
}
